binary_search_tree/tree.py

```diff
@@ -14,35 +14,128 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log n)
+    # Space Complexity: O(1)
     def add(self, key, value = None):
```

Comment:
👍  But the space complexity is O(log n) for a balanced tree because you're using recursion.